components/renku_data_services/message_queue/blueprints.py

```diff
@@ -0,0 +1,80 @@
+"""Search/reprovisioning blueprint."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from sanic import HTTPResponse, Request, json
+from sanic.response import JSONResponse
+from sqlalchemy.ext.asyncio import AsyncSession
+
+import renku_data_services.base_models as base_models
+from renku_data_services.authz.authz import Authz
+from renku_data_services.base_api.auth import authenticate, only_admins
+from renku_data_services.base_api.blueprint import BlueprintFactoryResponse, CustomBlueprint
+from renku_data_services.message_queue.core import reprovision
+from renku_data_services.message_queue.db import EventRepository, ReprovisioningRepository
+from renku_data_services.namespace.db import GroupRepository
+from renku_data_services.project.db import ProjectRepository
+from renku_data_services.users.db import UserRepo
+
+
+@dataclass(kw_only=True)
+class SearchBP(CustomBlueprint):
+    """Handlers for search."""
+
+    authenticator: base_models.Authenticator
+    session_maker: Callable[..., AsyncSession]
+    reprovisioning_repo: ReprovisioningRepository
+    event_repo: EventRepository
+    user_repo: UserRepo
+    group_repo: GroupRepository
+    project_repo: ProjectRepository
+    authz: Authz
+
+    def post(self) -> BlueprintFactoryResponse:
+        """Start a new reprovisioning."""
+
+        @authenticate(self.authenticator)
+        @only_admins
+        async def _post(request: Request, user: base_models.APIUser) -> HTTPResponse:
+            reprovisioning = await self.reprovisioning_repo.start()
+
+            request.app.add_task(
+                reprovision(
+                    session_maker=self.session_maker,
+                    requested_by=user,
+                    reprovisioning=reprovisioning,
+                    reprovisioning_repo=self.reprovisioning_repo,
+                    event_repo=self.event_repo,
+                    user_repo=self.user_repo,
+                    group_repo=self.group_repo,
+                    project_repo=self.project_repo,
+                    authz=self.authz,
+                ),
+                name=f"reprovisioning-{reprovisioning.id}",
+            )
+
+            return HTTPResponse(status=201)
+
+        return "/search/reprovision", ["POST"], _post
+
+    def get_status(self) -> BlueprintFactoryResponse:
+        """Get reprovisioning status."""
+
+        @authenticate(self.authenticator)
+        async def _get_status(_: Request, __: base_models.APIUser) -> JSONResponse | HTTPResponse:
+            active_reprovisioning = await self.reprovisioning_repo.get_active_reprovisioning()
+            return json({"active": bool(active_reprovisioning)}, 200)
+
+        return "/search/reprovision", ["GET"], _get_status
+
+    def delete(self) -> BlueprintFactoryResponse:
+        """Stop reprovisioning (if any)."""
```

Comment:
question: Do we really want this at all? Because if you start a reprovisioning but then stop it you leave the event queue in a weird inconsistent state. So maybe once you start a reprovisioning you have to wait for it to finish or error out.

Reply:
Some 2 cents: while a `delete` action may not be desirable, there should be a way to force a reprovisioning to be marked as errored. You cannot always ensure that a job will report back it has errored, so an admin needs to be able to mark it that way (and attempt a new one).

Reply:
Yeah the DELETE endpoint to stop a reprovisioning will stay.